src/edu/washington/escience/myria/operator/InMemoryOrderBy.java

```diff
@@ -193,8 +192,7 @@ public void sort() {
       for (int columnIdx = 0; columnIdx < getSchema().numColumns(); columnIdx++) {
         int sourceRow = indexes.get(rowIdx);
         int tupleIdx = table.getTupleIndexInContainingTB(sourceRow);
-        ReadableColumn hashTblColumn = table.getColumns(sourceRow)[columnIdx];
-        ans.put(columnIdx, hashTblColumn, tupleIdx);
+        ans.put(columnIdx, table.getColumn(columnIdx, sourceRow), tupleIdx);
```

Comment:
This usage suggests that the `TupleBatchBuffer`/`MutableTupleBuffer` API still isn't high-level enough: an ordinary client shouldn't have to worry about finding the correct `TupleBatch`, extracting the correct column from it, and calculating the correct row offset within that column. They should only have to specify source/destination row/column offsets and values. I think a symptom of this is that `getTupleIndexInContainingTB()` is a public API, which I think shouldn't be necessary (the implementations should be handling that calculation themselves).